business-registry-dashboard/app/pages/magicLink.vue

```diff
@@ -0,0 +1,112 @@
+<script setup lang="ts">
+import {
+  NrRequestActionCodes,
+  FilingTypes
+} from '@bcrs-shared-components/enums'
+
+const { t } = useI18n()
+const brdModal = useBrdModals()
+const route = useRoute()
+const affNav = useAffiliationNavigation()
+const { isAuthenticated } = useKeycloak()
+
+// Define eligible corporation types for incorporation
+const ELIGIBLE_CORP_TYPES = [
+  CorpTypes.BENEFIT_COMPANY,
+  CorpTypes.COOP,
+  CorpTypes.BC_CCC,
+  CorpTypes.BC_COMPANY,
+  CorpTypes.BC_ULC_COMPANY
+]
+
+definePageMeta({ order: 0 })
+
+/**
+ * Creates a new business from a Name Request
+ * @param business - Business object containing corpType and nameRequest
+ * @returns Promise<string> - The identifier of the newly created business
+ */
+const createBusinessFromNR = async (business: any) => {
+  const payload = {
+    filingType: FilingTypes.INCORPORATION_APPLICATION,
+    business
+  }
+
+  const filingResponse = await createNamedBusiness(payload)
+  return filingResponse.filing.business.identifier
+}
+
+/**
+ * Magic Link Processing Flow:
+ * 1. Verify user authentication
+ * 2. Extract and validate URL parameters (nrId, email, phone)
+ * 3. Fetch name request details
+ * 4. Verify business eligibility
+ * 5. Create business and redirect to dashboard
+ */
+onMounted(async () => {
+  // Redirect unauthenticated users to login page with current URL as redirect target
+  if (!isAuthenticated.value) {
+    const authWebUrl = useRuntimeConfig().public.authWebUrl
+    const redirectUrl = encodeURIComponent(window.location.href)
+    window.location.href = `${authWebUrl}/login?redirect=${redirectUrl}`
+  }
+
+  try {
+    // Extract query parameters from the URL
+    const { nrId, email, phone } = route.query as { nrId: string, email: string, phone: string }
+
+    // Ensure either email or phone is provided for contact
+    if (!email && !phone) {
+      brdModal.openMagicLinkModal(
+        t('error.magicLinkMissingContactInfo.title'),
+        t('error.magicLinkMissingContactInfo.description')
+      )
+      return
+    }
+
+    // Fetch name request details using provided credentials
+    const nameRequest = await fetchNameRequest(nrId, phone, email)
+
+    const business = {
+      corpType: { code: nameRequest.legalType },
+      nameRequest: {
+        ...nameRequest,
+        nrNumber: nameRequest.nrNum, // Transform nrNum to nrNumber
+        requestActionCd: nameRequest.request_action_cd // Transform request_action_cd to requestActionCd
+      }
+    }
+
+    // Verify business eligibility and proceed with incorporation
```

Comment:
Do you think the NR should be affiliated to the current account first?

Reply:
I think it's a super good idea to add that. So, I added a check to make sure the NR is affiliated to the account. This prevents now from let's say, BCREG0013 clicking on the magic link and it actually being processed when the NR is actually in BCREG0020...

Reply:
Oops, I meant that your code should affiliate the NR before creating the draft IA.

Reply:
Looks good.